packages/back-end/src/api/code-refs/postCodeRefs.ts

```diff
@@ -5,13 +5,55 @@ import { postCodeRefsValidator } from "back-end/src/validators/openapi";
 import {
   getFeatureCodeRefsByFeatures,
   upsertFeatureCodeRefs,
+  getAllCodeRefsForOrg,
 } from "back-end/src/models/FeatureCodeRefs";
+import { FeatureCodeRefsInterface } from "back-end/types/code-refs";
 
 export const postCodeRefs = createApiRequestHandler(postCodeRefsValidator)(
   async (req): Promise<PostCodeRefsResponse> => {
+    const { deleteMissing: deleteMissingString } = req.query;
     const { branch, repoName: repo } = req.body;
     const refsByFeature = groupBy(req.body.refs, "flagKey");
+    // convert deleteMissing to boolean
+    const deleteMissing = deleteMissingString === "true";
+    const allExistingCodeRefs: FeatureCodeRefsInterface[] = await getAllCodeRefsForOrg(
+      {
+        context: req.context,
+      }
+    );
+
+    const existingCodeRefsForRepoBranch = allExistingCodeRefs.filter(
+      (codeRef) => codeRef.repo === repo && codeRef.branch === branch
+    );
+
+    const existingFeatures = existingCodeRefsForRepoBranch.map(
+      (codeRef) => codeRef.feature
+    );
```

Comment:
Needs a dedupe to prevent unnecessary db calls. Maybe
```suggestion
    const existingFeatures = [...new Set(existingCodeRefsForRepoBranch.map(
      (codeRef) => codeRef.feature
    ))];
```